### ISSUE TITLE
Feature: Collapse comments

### ISSUE THREAD

**Issue by rishipuri** [thepracticaldev/dev.to#173]
# _Collapse comments_
### User Story
As a user I should be able to collapse comments.

**Comment by Zhao-Andy** [comment]
Thanks for the feedback. We're going to put this in our backlog for now.

**Comment by BornStellarMakesEternalLasting** [comment]
I love this idea. It would make reading comments quicker, like on reddit. I 100% back this idea!

**Comment by ravernkoh** [comment]
Highly support this idea. I always end up clicking on the arrow at the top right of a comment thinking that it would collapse the comment.

**Comment by pkfrank** [comment]
I'm also a big fan of this idea.  I think it's especially critical in longer threads (which are becoming increasingly common).

**Comment by spaceisstrange** [comment]
This is really needed now that Dev.to has become bigger. There are threads that are impossible to quickly navigate between parent comments because they have too many nested children comments. 

Another suggestion would be to change the arrow to show the menu and instead use three dots or another indicator. I've already got used to it and I'm still clicking it sometimes thinking it'll collapse the comment.

**Comment by Link2Twenty** [comment]
Changing each comment node at level 0 to a `<details>` block would do this.
We'd have to work out what the summary should read through (it's 'details' by default)

![details](https://user-images.githubusercontent.com/3534427/50778227-52cb6800-1295-11e9-92b4-3bcac4afd170.gif)

**Comment by rhymes** [comment]
@Link2Twenty woah! nice! The fact that you can just the let the browser deal with collapsing is awesome. A lot less code to deal with ;)

Probably still needs a polyfill for IE/Edge: https://caniuse.com/#search=details

**Comment by benhalpern** [comment]
Cool! I'm going to play with this today.

IE, I get. Edge not supporting this is fairly frustrating.

**Comment by rhymes** [comment]
@benhalpern well, worst case scenario you don't implement the feature for IE/Edge, the site isn't broken, "it's graceful degradation, folks" 🤣

**Comment by Link2Twenty** [comment]
When edge moves over to Blink these problems will be a thing of that past 😁

**Comment by Link2Twenty** [comment]
@benhalpern do you have access to usage data? What percentage of users use each browser?

**Comment by benhalpern** [comment]
Okay, I played around with this with a custom `<summary>`, let me know if folks like the way it looks here:

https://github.com/thepracticaldev/dev.to/pull/1484

![](https://camo.githubusercontent.com/05e651d58318b4d07f77b2812ca3de99243e3100/68747470733a2f2f636c2e6c792f3135373736363133313066332f53637265656e2532305265636f7264696e67253230323031392d30312d3037253230617425323030372e3539253230504d2e676966)
I also changed the "more info" menu to three dots for consistency with other parts of the site and so there were not two down-triangles.

Regarding Edge/IE, in playing around with a possible Polyfill I realized that not only will it not work in these browsers, the triangle won't show up either. Basically nothing will seem "broken" and this functionality just won't exist there. I think that's okay (this didn't seem totally obvious to me until I went through the exercise).

I'll sleep on this but possibly merge something tomorrow if we like it.
